turbine/src/sigverify_shreds.rs

```diff
@@ -146,25 +146,28 @@ fn run_shred_sigverify<const K: usize>(
     stats.num_packets += packets.iter().map(PacketBatch::len).sum::<usize>();
     stats.num_discards_pre += count_discards(&packets);
     // Repair shreds include a randomly generated u32 nonce, so it does not
-    // make sense to deduplicate them (i.e. they are not duplicate of any other
-    // shred).
+    // make sense to deduplicate the entire packet payload (i.e. they are not
+    // duplicate of any other packet.data(..)).
     // If the nonce is excluded from the deduper then false positives might
     // prevent us from repairing a block until the deduper is reset after
     // DEDUPER_RESET_CYCLE. A workaround is to also repair "coding" shreds to
     // add some redundancy but that is not implemented at the moment.
     // Because the repair nonce is already verified in shred-fetch-stage we can
-    // exclude repair shreds from the deduper.
+    // exclude repair shreds from the deduper, but we still need to pass the
+    // repair shred to the deduper to filter out duplicates from the turbine
+    // path once a shred is repaired.
+    // For backward compatibility we need to allow trailing bytes in the packet
+    // after the shred payload, but have to exclude them here from the deduper.
     stats.num_duplicates += thread_pool.install(|| {
         packets
             .par_iter_mut()
             .flatten()
             .filter(|packet| {
                 !packet.meta().discard()
-                    && !packet.meta().repair()
-                    && packet
-                        .data(..)
-                        .map(|data| deduper.dedup(data))
+                    && shred::wire::get_shred(packet)
+                        .map(|shred| deduper.dedup(shred))
                         .unwrap_or(true)
+                    && !packet.meta().repair()
```

Comment:
`get_shred()` will implicitly return `None` if `packet.meta().discard() == true`. So, we're arguably double checking here but also probably better for this file not to rely on the impl of `get_shred()` for correctness.

I guess if we really wanted to optimize out this out, we could add a unit test in this file to ensure that behavior ensues. Probably not worth it but thinking out loud here

Reply:
> probably better for this file not to rely on the impl of get_shred() for correctness.

yeah, that is my thinking.

> Probably noise in the grand scheme

yeah, I would be surprised if it has any noticeable impact either way.